lib/modules/manager/terraform/extractors/others/modules.ts

```diff
@@ -10,16 +10,16 @@ import { DependencyExtractor } from '../../base';
 import type { TerraformDefinitionFile } from '../../hcl/types';
 
 export const githubRefMatchRegex = regEx(
-  /github\.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?ref=(?<tag>.*)$/i
+  /github\.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?.*ref=(?<tag>.*)$/i
```

Comment:
what of the depth and ref are reversed? then all is captured as ref. we should probably tighten the regex some more to allow any param order.

Reply:
My reasoning was that you always want the `ref` at the end for readability, but you're right. It's still a valid URL even with `ref` before `depth`